core/json/json_path.rs

```diff
@@ -0,0 +1,137 @@
+use pest::Parser as P;
+use pest_derive::Parser;
+
+#[derive(Parser)]
+#[grammar = "json/json.pest"]
+#[grammar = "json/json_path.pest"]
+struct Parser;
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum PathElement {
```

Comment:
Could you also comment these using triple-slash comments? Each enum variant / struct field can also have its own triple-slash comment. IDEs pick those up which is helpful